src/WebJobs.Script/Rpc/ILanguageWorkerChannel.cs

```diff
@@ -5,12 +5,18 @@
 
 namespace Microsoft.Azure.WebJobs.Script.Rpc
 {
-    internal delegate ILanguageWorkerChannel CreateChannel(WorkerConfig conf, IObservable<FunctionRegistrationContext> registrations, int attemptCount);
+    public delegate ILanguageWorkerChannel CreateChannel(string language, IObservable<FunctionRegistrationContext> registrations, int attemptCount);
 
-    internal interface ILanguageWorkerChannel : IDisposable
+    public interface ILanguageWorkerChannel : IDisposable
     {
         string Id { get; }
 
         WorkerConfig Config { get; }
+
+        void RegisterFunctions(IObservable<FunctionRegistrationContext> functionRegistrations);
+
+        void SendFunctionEnvironmentReloadRequest();
```

Comment:
ReloadEnvironment?

Reply:
ReloadEnvironment is confusing because it assumes the action to be completed by the method. SendFunctionEnvironmentReloadRequest sends request to language worker. Reload is only complete when we get a response back from the language worker process. Event subscriptions take care of it.